calculators/_memoria_rating.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+$GLOBALS['rating_calculators']['_memoria_rating'] = [
+	'description' => '
+		by <a href="/player/125980">Memoria</a>: Rating with per-matchup rating calculation, v1
+		
+		-Only for 1v1, 2v2, and 3v3.
+  		-Games under 4 minutes don\'t count.
+		-Just like in the main rating, people start at 2000 rating,
+		 4000 rating difference = 90% win prediction, max 100 rating per match
+  		-Rating prediction is calculated for each matchup (1v1),
+		 then it\'s combined into a final prediction 
+		',
+	'default_rating' => 2000,
+	'fields' => [
+		'g.duration',
+	],
+	'main' => '_memoria_rating__main',
+	];
+
+function _memoria_rating__main(&$teams) {
+	$game_duration_seconds = floor($teams[0][0]['duration']/22.4);
+	if (count($teams[0]) != count($teams[1])) {
+		return;
+	}
+	
+	if ($game_duration_seconds < 240) {
+		return;
+	}
+		
+	$player_c = count($teams[0]);
+	$factor = 4000;
+	$ratings1 = [];
+	foreach ($teams[0] as $player) {
+        $ratings1[] = $player['rating'];
+    }
+	$ratings2 = [];
+	foreach ($teams[1] as $player) {
+        $ratings2[] = $player['rating'];
+    }
+	
+	$matchup_predictions = [];
+	$matchup_predictions_weight = [];
+	$matchup_predictions_weight_total = 0;
+	for ($i = 0; $i < $player_c; $i++) {
+		$matchup_predictions[$i] = 1-(round(100/(1 + 10**(($ratings1[$i]-$ratings2[$i])/$factor))))/100;
+		$matchup_predictions_weight[$i] = (abs(((0.5 - $matchup_predictions[$i])*3)))**1.6;
+		$matchup_predictions_weight_total += $matchup_predictions_weight[$i];
+	}
+	array_multisort($matchup_predictions_weight, SORT_DESC, $matchup_predictions);
+	
+	$prediction = 0;
+	$normalised_weight_mult_total = 0;
+	$j = 1;
```

Comment:
give this variable a name
what's this? I've been trying to understand what name it could have for like 5+ minutes, it is a counter/index of negative matchup predictions, but it is also some multiplier/denominator... I give up.

obviously not required, but highly recommended